sources/web/datalab/polymer/test/item-list-test.ts

```diff
@@ -436,5 +436,80 @@ describe('<item-list>', () => {
       assert(fifthDetailsContainer.getAttribute('hidden') != null,
           'fifth details container should be hidden');
     });
+
+    it('should hide the filter box by default', () => {
+      assert(testFixture.$.filterBox.offsetHeight === 0, 'filter box should not show by default');
+    });
+
+    it('shows/hides filter box when toggle is clicked', () => {
+      testFixture.$.filterToggle.click();
+      assert(testFixture.$.filterBox.offsetHeight > 0,
+          'filter box should show when toggle is clicked');
+
+      testFixture.$.filterToggle.click();
+      assert(testFixture.$.filterBox.offsetHeight === 0,
+          'filter box should hide when toggle is clicked again');
+    });
+
+    it('filters items when typing characters in the filter box', () => {
+      testFixture.$.filterToggle.click();
+      testFixture._filterString = '3';
+      Polymer.dom.flush();
+      const rows = testFixture.$.listContainer.querySelectorAll('.row');
+      assert(rows.length === 1, 'only one item has "3" in its name');
+      assert(rows[0].children[1].innerText === 'first column 3',
+          'filter should only return the third item');
+    });
+
+    it('shows all items when filter string is deleted', () => {
+      testFixture.$.filterToggle.click();
+      testFixture._filterString = '3';
+      Polymer.dom.flush();
+      testFixture._filterString = '';
+      Polymer.dom.flush();
+      const rows = testFixture.$.listContainer.querySelectorAll('.row');
+      assert(rows.length === 5, 'should show all rows after filter string is deleted');
+    });
+
+    it('filters items based on first column only', () => {
+      testFixture.$.filterToggle.click();
+      testFixture._filterString = 'second';
+      Polymer.dom.flush();
+      const rows = testFixture.$.listContainer.querySelectorAll('.row');
+      assert(rows.length === 0,
+          'should not show any rows, since no row has "second" in its first column');
+    });
+
+    it('ignores case when filtering', () => {
+      testFixture.$.filterToggle.click();
+      testFixture._filterString = 'FIRST';
+      Polymer.dom.flush();
+      const rows = testFixture.$.listContainer.querySelectorAll('.row');
+      assert(rows.length === 5, 'should show all rows, since filtering is case insensitive');
```

Comment:
Would be better to use a filter string that returns a subset of the list, to ensure that it is not simply ignoring all upper case in the filter string.

Reply:
Done.